pkg/splunk/enterprise/clustermanager.go

```diff
@@ -131,7 +132,14 @@ func ApplyClusterManager(ctx context.Context, client splcommon.ControllerClient,
 				return result, err
 			}
 		}
-		DeleteOwnerReferencesForResources(ctx, client, cr, &cr.Spec.SmartStore)
+
+		// Check if ClusterManager has any remaining references to other CRs, if so don't delete
+		err = checkCmRemainingReferences(ctx, client, cr)
```

Comment:
doesn't this apply to LM and MC too? since other CRs will reference LM and MC through licenseManagerRef and mcref?

Reply:
Yes it applies to LM and MC too. I have covered only the ClusterManager case in this PR.